lib/ip.go

```diff
@@ -37,34 +50,93 @@ func RandIP4List(n int) []net.IP {
 	return ips
 }
 
-// RandIP4Range returns a list of randomly generated IPv4 addresses within
-// starting and ending IPrange.
+// EvalIP4Range checks if the starting and ending IP range is valid or not.
+//
+// note: starting and ending IPs must be valid IPv4 string formats.
+func EvalIP4Range(
+	startIP string,
+	endIP string,
+) (IP4Range, error) {
+	startIPRaw := net.ParseIP(startIP).To4()
+	if len(startIPRaw) == 0 || len(startIPRaw) > net.IPv4len {
+		return IP4Range{}, errors.New("invalid range start IP")
+	}
 
-// note: starting and ending IPs should be in format of "1.1.1.1"
-func RandIP4Range(startIP, EndIP string) (net.IP, error) {
-	StartIPRaw := net.ParseIP(startIP).To4()
-	if len(StartIPRaw) == 0 || len(StartIPRaw) > net.IPv4len {
-		return nil, fmt.Errorf("range is Invalid")
+	startIPInt := binary.BigEndian.Uint32(startIPRaw.To4())
+	endIPRaw := net.ParseIP(endIP).To4()
+	if len(endIPRaw) == 0 || len(endIPRaw) > net.IPv4len {
+		return IP4Range{}, errors.New("invalid range end IP")
 	}
-	StartIPInt := binary.BigEndian.Uint32(StartIPRaw.To4())
-	EndIPRaw := net.ParseIP(EndIP).To4()
-	if len(EndIPRaw) == 0 || len(EndIPRaw) > net.IPv4len {
-		return nil, fmt.Errorf("range is Invalid")
+
+	endIPInt := binary.BigEndian.Uint32(endIPRaw.To4())
+
+	// ensure valid range
+	if startIPInt > endIPInt {
+		return IP4Range{}, fmt.Errorf("invalid range: %v > %v", startIP, endIP)
 	}
-	EndIPInt := binary.BigEndian.Uint32(EndIPRaw.To4())
-	if StartIPInt > EndIPInt {
-		return nil, fmt.Errorf("range is Invalid")
+	return IP4Range{
+		startIP: IP(startIPInt),
+		endIP:   IP(endIPInt),
+	}, nil
+}
+
+// EvalIP6Range checks if the starting and ending IP range is valid or not.
+//
+// note: starting and ending IPs must be valid IPv6 string formats.
+func EvalIP6Range(
```

Comment:
All of what was said about `EvalIP4Range` above will apply here too.

Reply:
We already have a `NewIP6Range` function in `ip6_range.go` file

Reply:
Then let's call it `NewIP6RangeInt`.